### ISSUE TITLE
Multiple outputs in ESSL3.0

### ISSUE THREAD

**Issue by mlomb** [LukasBanana/XShaderCompiler#69]
Input shader:
```
struct Output {
    float4 SV_A : SV_Target0;
    float4 SV_B : SV_Target1;
    float4 SV_C : SV_Target2;
};

Output main() {
    Output output;
    return output;
}
```
Output shader (ESSL300):
```
#version 300 es

out highp vec4 SV_Target0;
out highp vec4 SV_Target1;
out highp vec4 SV_Target2;

void main()
{
    
}
```
When compiling:
- 'SV_Target0' : must explicitly specify all locations when using multiple fragment outputs
- 'SV_Target1' : must explicitly specify all locations when using multiple fragment outputs
- 'SV_Target2' : must explicitly specify all locations when using multiple fragment outputs

In ESSL 3.0 specification (4.3.8.2):

> If there is more than one output, the location must be specified for all outputs.

So it should specify the location if there is more than one output (or the auto binding is enabled).

Sidenote:
The spec also says: 

> Fragment shaders cannot have input layout qualifiers.
> ...
> Vertex shaders cannot have output layout qualifiers.

But the autoBinding add the layout qualifier anyway.

Spec: https://www.khronos.org/registry/OpenGL/specs/es/3.0/GLSL_ES_Specification_3.00.pdf

**Comment by LukasBanana** [comment]
Thanks for the report. As you can see, I haven't work a lot with ESSL so far.
I know there are a lot of differences between GLSL and ESSL, but I haven't had the time yet to gather much experience with ESSL.

**Comment by LukasBanana** [comment]
Fixed with [56aee09](https://github.com/LukasBanana/XShaderCompiler/commit/56aee09c8bdad3912bac2638ac58bd720e39d204).

**Comment by mlomb** [comment]
I have problems trying to compile the same shader again, seems like it doesn't work when autoBinding is disabled. The location should be there even if the autoBinding is disabled, at least for ESSL3.0/1/2.

Also, doing some tests with the autoBinding enabled I found that the uniform blocks doesn't support binding:
```
layout(std140, row_major, binding = 0) uniform UBO
{
    highp mat4 some_matrix;
};
```
- 'binding' : invalid layout qualifier: not supported
- 'binding' : invalid layout qualifier: only valid when used with opaque types or blocks

**Comment by LukasBanana** [comment]
Feel free re-open this issue. I'll take care of it tomorrow.

**Comment by mlomb** [comment]
I can't reopen the issue if the owner (you) close it.

**Comment by LukasBanana** [comment]
I maybe ran over this a little fast, so I'll take a closer look at it as soon as I can.
Thanks and kind regards,
Lukas

**Comment by LukasBanana** [comment]
Should be fixed now with [c268e07](https://github.com/LukasBanana/XShaderCompiler/commit/c268e073bbc7872e1262f90a00d1aedf00c37585).
The `binding` qualifier is only generated if `-EB` is enabled and the output ESSL version is at least 310.
Let me know if there's still something wrong on this.
Lukas

**Comment by mlomb** [comment]
Everything seems to be working fine :)
